src/main/java/org/owasp/html/AutoCloseableExtensibleHtmlStreamRenderer.java

```diff
@@ -0,0 +1,96 @@
+package org.owasp.html;
+
+import javax.annotation.WillCloseWhenClosed;
+import java.io.Closeable;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Set;
+
+final class AutoCloseableExtensibleHtmlStreamRenderer extends ExtensibleHtmlStreamRenderer
+  // This is available on JDK6 and makes this class extend AutoCloseable.
+  implements Closeable {
+  private final Object closeable;
+
+  private static final Class<?> CLASS_AUTO_CLOSEABLE;
+
+  static {
+    Class<?> classAutoCloseable = null;
+    for (Class<?> superInterface : Closeable.class.getInterfaces()) {
+      if ("java.lang.AutoCloseable".equals(superInterface.getName())) {
```

Comment:
Is there any method like CLASS_AUTO_CLOSEABLE.getName() or something to compare?

Reply:
this class is referenced from existing OWASP class, with as much less modification posible